src/Functions/CMakeLists.txt

```diff
@@ -13,7 +13,6 @@ set(DBMS_FUNCTIONS
     multiIf.cpp
     multiMatchAny.cpp
     checkHyperscanRegexp.cpp
-    array/has.cpp
```

Comment:
At this point, if we are linking `clickhouse_functions` everywhere, do we even need these blocks that moves files or headers to `dbms`? Maybe it'd be simpler to just use clickhouse_functions eveywhere

Reply:
i spent quite some time refusing to check for this, but yes

https://github.com/ClickHouse/ClickHouse/blob/8b20e9f505f5faaac3690c24c2f4f69dd6bbbe80/src/Functions/tests/gtest_ternary_logic.cpp#L11-L12

Reply:
Solution: Remove this file

Reply:
i made an attempt to fix it, but there were too many dependencies from the source file that i didn't want to move to the header